flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/connector/datagen/table/RandomGeneratorVisitor.java

```diff
@@ -503,10 +524,14 @@ private static RandomGenerator<byte[]> getRandomBytesGenerator(int length) {
         return new RandomGenerator<byte[]>() {
             @Override
             public byte[] next() {
-                byte[] arr = new byte[length];
+                byte[] arr = new byte[getVariableLengthFieldRealLen(length, varLen)];
                 random.getRandomGenerator().nextBytes(arr);
                 return arr;
             }
         };
     }
+
+    private static int getVariableLengthFieldRealLen(int length, boolean varLen) {
+        return varLen ? (new Random().nextInt(length)) + 1 : length;
```

Comment:
we can avoid per-call new instance here by using a lazy initialized `RandomGenerator` member variable(and it also support lower bound so that `+ 1` can be omitted)

Reply:
What about using static ThreadLocalRandom.current() provided by java instead of per-call new instance? I found that DecimalDataRandomGenerator#next also uses such method to generate random data.
```
    public DecimalData next() {
        if (nullRate == 0f || ThreadLocalRandom.current().nextFloat() > nullRate) {
            BigDecimal decimal =
                    new BigDecimal(
                            ThreadLocalRandom.current().nextDouble(min, max),
                            new MathContext(precision, RoundingMode.DOWN));
            return DecimalData.fromBigDecimal(decimal, precision, scale);
        }
        return null;
    }
```